bindata/network/openshift-sdn/sdn-ovs.yaml

```diff
@@ -50,8 +50,9 @@ spec:
             # the character "-" but there may be more in the future.
             echo "${1//-/_2d}"
           }
+
           # Check to see if ovs is provided by the node:
-          if [[ -f '/host/usr/local/bin/configure-ovs.sh' ]]; then
+          if [[ -L '/host/etc/systemd/system/network-online.target.wants/ovs-configuration.service' ]]; then
```

Comment:
as a fail-safe is it possible with dbus to try to start `ovs-vswitchd` regardless.  presumably `org.freedesktop.systemd1.Unit.Start` is a no-op if `ovs-vswitchd` is already started.

Reply:
We tried that originally and the short answer is "it's more complicated than we thought, error-prone, and not possible to do yet". Mainly due to containers having different UID, PID space, and stuff like that.

Reply:
the dbus API seems to work inside the `ovs` container, it starts a stopped `ovs-vswitchd` and doesn't affect a started `ovs-vswitchd`

```bash
gdbus call --system --dest org.freedesktop.systemd1 --object-path /org/freedesktop/systemd1/unit/ovs_2dvswitchd_2eservice --method org.freedesktop.systemd1.Unit.Start "replace"
```

Reply:
we dont want to start ovs here, we just want to detect if it is running in systemd. Through upgrade process we need container ovs to still work.

Reply:
Ah, I meant after this line, once we detect `ovs-configuration.service` we could add a failsafe `ovs-vswitchd` start.

If `ovs-vswitchd` is already running then `systemd` won't do anything, but other-wise we have a failsafe.

We seem to already have a failsafe start of `ovsdb-server` on line 63.

Reply:
```diff
--- a/bindata/network/openshift-sdn/sdn-ovs.yaml
+++ b/bindata/network/openshift-sdn/sdn-ovs.yaml
@@ -38,12 +38,12 @@ spec:

           # systemctl cannot be used in a separate PID namespace to reach
           # the systemd running in PID 1. Therefore we need to use the dbus API
-          systemctl_restart(){
+          systemctl_dbus(){
             gdbus call \
               --system \
               --dest org.freedesktop.systemd1 \
-              --object-path /org/freedesktop/systemd1/unit/"$(svc_encode_name ${1})"_2eservice \
-              --method org.freedesktop.systemd1.Unit.Restart "replace"
+              --object-path /org/freedesktop/systemd1/unit/"$(svc_encode_name ${2})"_2eservice \
+              --method org.freedesktop.systemd1.Unit."${1}" "replace"
           }
           svc_encode_name(){
             # systemd encodes some characters, so far we only need to encode
@@ -53,13 +53,14 @@ spec:
           # Check to see if ovs is provided by the node:
           if [[ -f '/host/usr/local/bin/configure-ovs.sh' ]]; then
             echo "openvswitch is running in systemd"
+            systemctl_dbus Start ovs-vswitchd
             # In some very strange corner cases, the owner for /run/openvswitch
             # can be wrong, so we need to clean up and restart.
             ovs_uid=$(chroot /host id -u openvswitch)
             ovs_gid=$(chroot /host id -g openvswitch)
             chown -R "${ovs_uid}:${ovs_gid}" /run/openvswitch
             if [[ ! -S /run/openvswitch/db.sock ]]; then
-              systemctl_restart ovsdb-server
+              systemctl_dbus Restart ovsdb-server
             fi
             # Don't need to worry about restoring flows; this can only change if we've rebooted
             exec tail -F /host/var/log/openvswitch/ovs-vswitchd.log /host/var/log/openvswitch/ovsdb-server.log
```